src/collector.rs

```diff
@@ -51,6 +51,19 @@ impl Collector {
         self.raw.get(command).unwrap().clone()
     }
 
+    // returns a string representation of the first capture group of the regex
+    pub fn get_regex_from_command(&mut self, command: &str, regex: &str) -> Option<String> {
+        let re = Regex::new(regex).unwrap();
+        let out = self.run_command(command);
+        let output = String::from_utf8(out.stdout).unwrap();
```

Comment:
some commands (like `gcc -v`) will output on `stderr`, may want to make this an option, or parse both? probably option?